scripts/populate_ingest/populate_ingest_from_nxn.py

```diff
@@ -0,0 +1,178 @@
+from convert.europe_pmc import EuropePmcConverter
+from data_entities.nxn_db import NxnDatabase
+from services.europe_pmc import EuropePmc
+from services.nxn_db import NxnDatabaseService
+from services.ingest import QuickIngest
+
+
+# INGEST_URL = "https://api.ingest.archive.data.humancellatlas.org"
+INGEST_URL = "http://localhost:8080"
+
+# utility functions
+def reformat_title(title: str) -> str:
+    return re.sub("\W", "", title).lower().strip()
+
+def get_distance_metric(title1: str,title2: str):
+    if not all([title1, title2]):
+        return 0
+    max_len = max(len(title1), len(title2))
+    dist_metric = 100-(Levenshtein.distance(title1, title2)/max_len)*100
+    return dist_metric
+
+def get_file_content(file_path):
+    with open(file_path, "r") as file_path:
+        return file_path.read()
+
+def prepare_logging():
+    logging.basicConfig(filename='nxn_db.log', format='%(asctime)s - %(levelname)s - %(message)s', level=logging.INFO)
+
+    def handle_exception(exc_type, exc_value, exc_traceback):
+        if issubclass(exc_type, KeyboardInterrupt):
+            sys.__excepthook__(exc_type, exc_value, exc_traceback)
+            return
+
+        logging.error("Exception", exc_info=(exc_type, exc_value, exc_traceback))
+
+    sys.excepthook = handle_exception
+
+class Populate:
+    def __init__(self, url, token, write):
+        self.write = write
+        self.ingest_api = QuickIngest(url, token=get_file_content(token))
+        self.ingest_data = [data.get("content") for data in self.ingest_api.get_all_projects()]
+        self.nxn_data = NxnDatabase(NxnDatabaseService.get_data())
+        self.europe_pmc = EuropePmc()
+        self.publication_converter = EuropePmcConverter()
+        self.ingest_schema = self.ingest_api.get_schemas(high_level_entity="type",
+                                                   domain_entity="project",
+                                                   concrete_entity="project")[0]['_links']['json-schema']['href']
+
+    def __compare_on_doi__(self):
+        ingest_data_pubs = list(itertools.chain.from_iterable(
+            [data.get("publications") for data in self.ingest_data if data.get("publications")]))
+        ingest_data_pub_doi = {pub.get("doi") for pub in ingest_data_pubs if pub.get("doi")}
+        ingest_data_pub_urls = {pub.get("url") for pub in ingest_data_pubs if pub.get("url")}
+        ingest_data_pre_doi = {url.split('doi.org/')[1] for url in ingest_data_pub_urls if 'doi.org/' in url}
+
+        filter_doi = (self.nxn_data.get_values('DOI') | self.nxn_data.get_values('bioRxiv DOI')) - (
+                    ingest_data_pub_doi | ingest_data_pre_doi)
+        self.nxn_data.data = [row for row in self.nxn_data.data if self.nxn_data.get_value(row, 'DOI') in filter_doi or
+                              self.nxn_data.get_value(row, 'bioRxiv DOI') in filter_doi]
+
+    def __compare_on_accession__(self):
+        ingest_data_accessions = set(itertools.chain.from_iterable(
```

Comment:
This seems very repetitive. Consider adding the IDs to a list and mapping the elements.